app/src/org/commcare/activities/UpdateActivity.java

```diff
@@ -49,6 +54,12 @@ protected void onCreate(Bundle savedInstanceState) {
 
         boolean isRotation = savedInstanceState != null;
         setupUpdateTask(isRotation);
+
+        proceedAutomatically = getIntent().getBooleanExtra(
+                RefreshToLatestBuildActivity.KEY_FROM_LATEST_BUILD_ACTIVITY, false);
+        if (proceedAutomatically) {
+            startUpdateCheck();
+        }
```

Comment:
You shouldn't need to call `startUpdateCheck()` here because it should have already been called in `setupUpdateTask()`